pass/pass_test.go

```diff
@@ -8,56 +11,88 @@ import (
 )
 
 func TestPassHelper(t *testing.T) {
-	helper := Pass{}
-
 	creds := &credentials.Credentials{
 		ServerURL: "https://foobar.docker.io:2376/v1",
 		Username:  "nothing",
 		Secret:    "isthebestmeshuggahalbum",
 	}
 
+	helper := Pass{}
 	_ = helper.CheckInitialized()
 
-	helper.Add(creds)
+	if err := helper.Add(creds); err != nil {
+		t.Fatal(err)
+	}
+
+	u, s, err := helper.Get(creds.ServerURL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if u != creds.Username {
+		t.Fatalf("invalid username %s", u)
+	}
+	if s != creds.Secret {
+		t.Fatalf("invalid secret: %s", s)
+	}
+
+	if err = helper.Delete(creds.ServerURL); err != nil {
+		t.Fatal(err)
+	}
+	if _, _, err = helper.Get(creds.ServerURL); !credentials.IsErrCredentialsNotFound(err) {
+		t.Fatalf("expected credentials not found, actual: %v", err)
+	}
+}
+
+func TestPassHelperList(t *testing.T) {
+	creds := []*credentials.Credentials{
+		{
+			ServerURL: "https://foobar.docker.io:2376/v1",
+			Username:  "foo",
+			Secret:    "isthebestmeshuggahalbum",
+		},
+		{
+			ServerURL: "https://foobar.docker.io:2375/v1",
+			Username:  "bar",
+			Secret:    "isthebestmeshuggahalbum",
+		},
+	}
 
-	creds.ServerURL = "https://foobar.docker.io:9999/v2"
-	helper.Add(creds)
+	helper := Pass{}
+	_ = helper.CheckInitialized()
+
+	for _, cred := range creds {
+		if err := helper.Add(cred); err != nil {
+			t.Fatal(err)
+		}
+	}
 
 	credsList, err := helper.List()
 	if err != nil {
 		t.Fatal(err)
 	}
-
 	for server, username := range credsList {
-		if !(strings.Contains(server, "2376") ||
-			strings.Contains(server, "9999")) {
-			t.Fatalf("invalid url: %s", creds.ServerURL)
+		if !(strings.HasSuffix(server, "2376/v1") || strings.HasSuffix(server, "2375/v1")) {
+			t.Fatalf("invalid url: %s", server)
 		}
-
-		if username != "nothing" {
+		if !(username == "foo" || username == "bar") {
 			t.Fatalf("invalid username: %v", username)
 		}
 
 		u, s, err := helper.Get(server)
 		if err != nil {
 			t.Fatal(err)
 		}
-
 		if u != username {
 			t.Fatalf("invalid username %s", u)
 		}
-
 		if s != "isthebestmeshuggahalbum" {
 			t.Fatalf("invalid secret: %s", s)
 		}
 
-		err = helper.Delete(server)
-		if err != nil {
+		if err = helper.Delete(server); err != nil {
 			t.Fatal(err)
 		}
-
-		_, _, err = helper.Get(server)
-		if !credentials.IsErrCredentialsNotFound(err) {
+		if _, _, err = helper.Get(server); !credentials.IsErrCredentialsNotFound(err) {
```

Comment:
if we're making these local, it should define a new variable

```suggestion
		if _, _, err := helper.Get(server); !credentials.IsErrCredentialsNotFound(err) {
```

(same for the other ones); not 100% sure if it's much more readable though 😅